src/vs/editor/contrib/stickyScroll/browser/stickyScrollProvider.ts

```diff
@@ -96,10 +96,44 @@ export class StickyLineCandidateProvider extends Disposable {
 		}
 	}
 
+	private customBinarySearch(model: readonly StickyOutlineElement[], startLine: number) {
```

Comment:
There is [binarySearch](https://github.com/microsoft/vscode/blob/e7f76d71041ddb7e73bdf0cf258b811902dc4c8c/src/vs/base/common/arrays.ts#L72) and [binarySearch2](https://github.com/microsoft/vscode/blob/e7f76d71041ddb7e73bdf0cf258b811902dc4c8c/src/vs/base/common/arrays.ts#L91) and you have a very strong for not using any of those ;-)